src/charm.py

```diff
@@ -138,61 +110,33 @@ def _on_certificate_available(self, event: CertificateAvailableEvent) -> None:
             "csr": event.certificate_signing_request,
         }
         if self._certificate_is_stored:
-            certificate_secret = self.model.get_secret(label=CERTIFICATE_SECRET_LABEL)
+            certificate_secret = self.model.get_secret(label=self._get_certificate_secret_label())
             certificate_secret.set_content(content=certificate_secret_content)
         else:
-            self.app.add_secret(
+            self.unit.add_secret(
                 content=certificate_secret_content,
-                label=CERTIFICATE_SECRET_LABEL,
+                label=self._get_certificate_secret_label(),
             )
         logger.info(f"New certificate is stored: {event.certificate}")
-        self.unit.status = ActiveStatus()
-
-    def _on_config_changed(self, event: ConfigChangedEvent) -> None:
-        """Validates config and requests a new certificate.
-
-        Args:
-            event: Juju event.
-        """
-        if not self.unit.is_leader():
-            return
-        if not self._config_subject:
-            self.unit.status = BlockedStatus("Config `subject` must be set.")
-            return
-        if not self._private_key_is_stored:
-            self.unit.status = WaitingStatus(
-                "Waiting for private key and password to be generated."
-            )
-            event.defer()
-            return
-        if not self._certificates_relation_created:
```

Comment:
Why don't we want to block waiting for the relation?

Reply:
Certificate status can be figured out through the `get-certificate` juju action anyway so I didn't think we needed to look at the status too and get the incomplete picture there.

Reply:
Got it :)